### ISSUE TITLE
const gene values

### ISSUE THREAD

**Issue by snesic** [broadinstitute/tensorqtl#9]
Hi,

Here is a code that returns an error (cannot perform reduction function max on tensor with no elements because the operation does not have an identity) when a gene contains the same value for all samples. If you uncomment  ph.loc['gene_1', 'sample_1'] = 6 it finishes successfully.

```
# Const gene
import pandas as pd
from tensorqtl import cis
from numpy.random import randint

samples = 10
snps = 2
genes = 2

print('Start: Const gene')

gt = pd.DataFrame(randint(-1, 3, size=(snps,samples)), 
                  columns=['sample_{}'.format(i) for i in range(1,samples+1)], 
                  index=['snp_{}'.format(i) for i in range(1,snps+1)])

gt_pos = pd.DataFrame({'pos' : list(range(1,snps+1)),
                      'chrom': '1'},
                      index = gt.index)

ph = pd.DataFrame(randint(5, 15, size=(genes, samples)), 
                  columns = gt.columns, 
                  index = ['gene_{}'.format(i) for i in range(1,genes+1)])

ph.loc['gene_1'] = 5
#ph.loc['gene_1', 'sample_1'] = 6
print(ph.loc['gene_1'])

covariates_df = pd.DataFrame(1, index=['C'], columns=ph.columns).T

#genes_pos chr tss
ph_pos = pd.DataFrame({'chr' : ['1' ,'1'], 'tss' : [100,1000]})
ph_pos.index = ph.index

cis_df = cis.map_cis(gt, gt_pos, ph, ph_pos, covariates_df)
```
Thanks!

**Comment by francois-a** [comment]
Thanks for the example! I had only tested this edge case for the nominal mapping and missed this. Fixed in 60ffdd4f887f430504cff67b2ff6948cee3c5152 -- constant phenotypes are now filtered out at the beginning.

**Comment by snesic** [comment]
Perfect, thanks!
